module/VuFind/src/VuFind/DigitalContent/OverdriveConnector.php

```diff
@@ -0,0 +1,1557 @@
+use Zend\Log\LoggerAwareInterface;
+use ZfcRbac\Service\AuthorizationServiceAwareInterface;
+use ZfcRbac\Service\AuthorizationServiceAwareTrait;
+
+/**
+ * OverdriveConnector
+ *
+ * Class responsible for connecting to the Overdrive API
+ *
+ * @category VuFind
+ * @package  DigitalContent
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public
+ *           License
+ * @link     https://vufind.org/wiki/development Wiki
+ * @todo     provide option for autocheckout by default in config
+ *       allow override for cover display using other covers
+ *       provide option for not requiring email for holds
+ *       provide option for giving users option for every hold
+ *       provide option for asking about autocheckout for every hold
+ *       provide config options for how to handle patrons with no access to OD
+ */
+class OverdriveConnector implements LoggerAwareInterface,
+    AuthorizationServiceAwareInterface, \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+    use AuthorizationServiceAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+    use KeyGeneratorTrait;
+
+    /**
+     * Session Container
+     *
+     * @var \Zend\Session\Container
+     */
+    protected $sessionContainer;
+
+    /**
+     * Record Config
+     *
+     * Main configurations
+     *
+     * @var \Zend\Config\Config
+     */
+    protected $recordConfig;
+
+    /**
+     * Record Config
+     *
+     * Overdrive configurations
+     *
+     * @var \Zend\Config\Config
+     */
+    protected $mainConfig;
+
+    /**
+     * ILS Authorization
+     *
+     * @var \VuFind\Auth\ILSAuthenticator
+     */
+    protected $ilsAuth;
+
+    /**
+     * HTTP Client
+     *
+     * Client for making calls to the API
+     *
+     * @var \Zend\Http\Client
+     */
+    protected $client;
+
+    /**
+     * Cache for storing ILS data temporarily (e.g. patron blocks)
+     *
+     * @var StorageInterface
+     */
+    protected $cache = null;
+
+    /**
+     * Container
+     *
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * Constructor
+     *
+     * @param \Zend\Config\Config           $mainConfig   VuFind main conf
+     * @param \Zend\Config\Config           $recordConfig Record-specific conf
+     *                                                    file
+     * @param ContainerInterface            $container    The whole enchilada
+     * @param \VuFind\Auth\ILSAuthenticator $ilsAuth      ILS Authenticator
+     */
+    public function __construct(
+        $mainConfig,
+        $recordConfig,
+        $container,
+        $ilsAuth
+    ) {
+        $this->debug("SolrOverdrive Connector");
+        $this->mainConfig = $mainConfig;
+        $this->recordConfig = $recordConfig;
+        $this->container = $container;
```

Comment:
I agree and that's the way it was.  The reason I changed it is because we are trying to make more use of the "console" functionality.  ex: a cron job that runs through users' saved searches to find new titles and notify them.  If I have the session controller in there, it chokes when trying to instantiate it in the console context.  Maybe there is a better way to do it...

Reply:
You can do a check: `if (PHP_SAPI !== 'cli') { ... }` to skip logic that won't work in console mode. Perhaps you can wrap the problem logic in the connector and/or the factory to work around the problem. Does that help?

Reply:
OK I fixed this.  It's working OK for now.  I haven't been able to test it in the console environment yet.  But I can do that later.
also, what is the rationale for not injecting the container?  Does that make it brittle somehow?